lib/utils/stream.js

```diff
@@ -12,16 +12,19 @@
 var Stream = function() {
   this.init = function() {
     var listeners = {};
+    var sentinels = {};
     /**
      * Add a listener for a specified event type.
      * @param type {string} the event name
      * @param listener {function} the callback to be invoked when an event of
      * the specified type occurs
      */
-    this.on = function(type, listener) {
+    this.on = function(type, listener, sentinel) {
```

Comment:
I think we should try very hard to stick with a node-compatible API here. First, we keep discussing using browserify's Stream implementation and this would make that difficult. Secondly, it may be confusing to people learning the project to see something that looks *really really* like a Stream but has a subtly different API.